### PR TITLE
Added docs for configurable variables

### PR DESCRIPTION
**Pull request by keyluck**
# Proposed Changes

Added RFC for configurable variables

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

- [ ] My changes generate no new warnings
- [ ] Logging has been added where applicable (Error handling, user-driven events, API requests/responses)

- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] Tests meet the test coverage threshold

## Reviewer Checklist

- [ ] Is the code easy to understand?
- [ ] Are the changes well documented?

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?

- [ ] Should any logging or debugging information be added or removed?

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining how you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/RFC/configurable-variables.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# [RFC] Configurable Variables
+
+**Summary**:
+
+As a requirement for our continuous integration process, we need to deploy the Embark application to multiple environments without rebuilding images for each environment. The purpose of this RFC is to identify the set of configurable variables Embark that can be managed independently of the deployment definition.
```
Comment (mhyder1) [inline docs/RFC/configurable-variables.md:5]: I don't think we need the word 'Embark' in the second sentence starting with "The purpose of this RFC"

File: docs/RFC/configurable-variables.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+## Findings
+
+### Building the application
+
+There are two scripts located in `package.json` that are used to build the frontend and backend assets:
+
+- `"build backend": "yarn workspace @department-of-veterans-affairs/backend backstage-cli backend:bundle"`
+This script builds the backend bundle and copies the output to `packages/backend/dist`.
+
+- `"build frontend": "cp app-config.yaml app-config.dev.yaml packages/app/ && yarn workspace @department-of-veterans-affairs/app backstage-cli app:build --config app-config.yaml --config app-config.dev.yaml"`
```
Comment (rianfowler) [inline docs/RFC/configurable-variables.md:29]: Are these the files that are copied into the image or are they merged as part of the build process? My understanding is that the application start command merges the configs. 
https://github.com/department-of-veterans-affairs/lighthouse-backstage/blob/a902e93741335402239d5e57a98af0afe6763ff8/Dockerfile.backend#L18
In the docker file we're still passing both configs in. 

How do you think we could have a single image that could run in any environment with different app configs for each environment?

**Review by rianfowler — APPROVED** [review]

File: docs/RFC/configurable-variables.md (hunk 2 above)
Comment (keyluck) [reply docs/RFC/configurable-variables.md:29]: When multiple configs are used Backstage gives higher priority to the configurations to the right. So if we replaced app-config.dev.yaml with app-config.production.yaml, the image would use the app-config.production.yaml by default. We then can use that same image to deploy to any nonprod environment by using environment variable overrides with the prefix `APP_CONFIG` (i.e. APP_CONFIG_app_baseUrl, APP_CONFIG_backend_baseUrl, etc.). This approach also includes a configMap, created only for nonprod environments, with the helm charts to map those overrides for the frontend and backend.

File: docs/RFC/configurable-variables.md (hunk 1 above)
Comment (keyluck) [reply docs/RFC/configurable-variables.md]: Woops, deleted that

**Pull request merged**